bindings/sftp/sftp.go

```diff
@@ -0,0 +1,314 @@
+	}
+
+	path, err := metadata.getPath()
+	if err != nil {
+		return nil, fmt.Errorf("sftp binding error: %w", err)
+	}
+
+	files, err := sftp.sftpClient.ReadDir(path)
+	if err != nil {
+		return nil, fmt.Errorf("sftp binding error: error read dir: %s %w", path, err)
+	}
+
+	var resp []listResponse
+
+	for _, file := range files {
+		resp = append(resp, listResponse{
+			FileName:    file.Name(),
+			IsDirectory: file.IsDir(),
+		})
+	}
+
+	jsonResponse, err := json.Marshal(resp)
+	if err != nil {
+		return nil, fmt.Errorf("sftp binding error: cannot marshal list to json: %w", err)
+	}
+
+	return &bindings.InvokeResponse{
+		Data: jsonResponse,
+	}, nil
+}
+
+func (sftp *Sftp) get(_ context.Context, req *bindings.InvokeRequest) (*bindings.InvokeResponse, error) {
+	metadata, err := sftp.metadata.mergeWithRequestMetadata(req)
+	if err != nil {
+		return nil, fmt.Errorf("sftp binding error: error merging metadata: %w", err)
+	}
+
+	path, err := metadata.getPath()
+	if err != nil {
+		return nil, fmt.Errorf("sftp binding error: %w", err)
+	}
+
+	file, err := sftp.sftpClient.Open(path)
+	if err != nil {
+		return nil, fmt.Errorf("sftp binding error: error open file: %s %w", path, err)
+	}
+
+	b, err := io.ReadAll(file)
+	if err != nil {
+		return nil, fmt.Errorf("sftp binding error: error read file: %s %w", path, err)
+	}
+
+	return &bindings.InvokeResponse{
+		Data: b,
+	}, nil
+}
+
+func (sftp *Sftp) delete(_ context.Context, req *bindings.InvokeRequest) (*bindings.InvokeResponse, error) {
+	metadata, err := sftp.metadata.mergeWithRequestMetadata(req)
+	if err != nil {
+		return nil, fmt.Errorf("sftp binding error: error merging metadata: %w", err)
+	}
+
+	path, err := metadata.getPath()
+	if err != nil {
+		return nil, fmt.Errorf("sftp binding error: %w", err)
+	}
+
+	err = sftp.sftpClient.Remove(path)
+	if err != nil {
+		return nil, fmt.Errorf("sftp binding error: error remove file: %s %w", path, err)
+	}
+
+	return nil, nil
+}
+
+func (sftp *Sftp) Invoke(ctx context.Context, req *bindings.InvokeRequest) (*bindings.InvokeResponse, error) {
+	switch req.Operation {
+	case bindings.CreateOperation:
+		return sftp.create(ctx, req)
+	case bindings.GetOperation:
+		return sftp.get(ctx, req)
+	case bindings.DeleteOperation:
+		return sftp.delete(ctx, req)
+	case bindings.ListOperation:
+		return sftp.list(ctx, req)
+	default:
+		return nil, fmt.Errorf("unsupported operation %s", req.Operation)
+	}
+}
+
+func (sftp *Sftp) Close() error {
+	return sftp.sftpClient.Close()
+}
+
+func (metadata sftpMetadata) getPath() (path string, err error) {
+	path = sftpClient.Join(metadata.RootPath, metadata.FileName)
+
+	if path == "" {
+		err = fmt.Errorf("required metadata rootPath or fileName missing")
+	}
+
+	return
+}
+
+// Helper to merge config and request metadata.
+func (metadata sftpMetadata) mergeWithRequestMetadata(req *bindings.InvokeRequest) (sftpMetadata, error) {
+	merged := metadata
+
+	if val, ok := req.Metadata[metadataRootPath]; ok && val != "" {
```

Comment:
Using `req.Metadata` with direct access makes the lookup case-sensitive. Ideally we should always use [metadata.GetMetadataProperty](https://pkg.go.dev/github.com/dapr/kit@v0.13.2/metadata#GetMetadataProperty) from dapr/kit which is case-insensitive